### ISSUE TITLE
Wi-Fi is losing connection

### ISSUE THREAD

**Issue by laurijamsa** [ruuvi/ruuvi.gateway_esp.c#340]
On my Gateways the Wi-Fi connection still works not really reliably. Usually when I'm able to configure it successfully, it works, but this behaviour is not something I've seen before I think. This may be related to my Wi-Fi network somehow, but not confirmed. My other devices work ok on this network. Will try to reproduce this issue with another Wi-Fi network.

```I (706942) http: [adv_post_task] HTTP POST to URL=https://network.ruuvi.com/record, DATA:
{
        "data": {
                "coordinates":  "",
                "timestamp":    "1630260478",
                "nonce":        "1862047883",
                "gw_mac":       "EE:39:AD:AB:D4:E2",
                "tags": {
                        "F8:78:29:D9:1B:80":    {
                                "rssi": -77,
                                "timestamp":    "1630260477",
                                "data": "0201061BFF9904050FE35A50C83B040CFF100024AA96137650F87829D91B80"
                        },
                        "DF:AA:90:34:69:05":    {
                                "rssi": -82,
                                "timestamp":    "1630260478",
                                "data": "0201061BFF9904050F7A544BC7AE001CFFCC03FCB3D6C3A4CCDFAA90346905"
                        },
                        "F0:4C:E2:AA:3F:13":    {
                                "rssi": -78,
                                "timestamp":    "1630260473",
                                "data": "0201061BFF9904050FB053AAC867000CFFC40420B236CE4DAFF04CE2AA3F13"
                        },
                        "F6:07:52:34:28:C6":    {
                                "rssi": -75,
                                "timestamp":    "1630260477",
                                "data": "0201061BFF9904050FF1548DC80A03E4FF08FFE4ACB621B817F607523428C6"
                        },
                        "CF:C6:30:48:71:CE":    {
                                "rssi": -70,
                                "timestamp":    "1630260477",
                                "data": "0201061BFF990405104D4D881133004CFFB003E89F967959B1CFC6304871CE"
                        },
                        "E5:06:6D:26:1F:9C":    {
                                "rssi": -78,
                                "timestamp":    "1630260477",
                                "data": "0201061BFF990405107B4E6B167F0014FFE00410AEF67B59E7E5066D261F9C"
                        },
                        "D7:FB:DB:12:9C:7D":    {
                                "rssi": -89,
                                "timestamp":    "1630260472",
                                "data": "0201061BFF99040511AF556CC80903F8FEF4FFB4AD965C6F09D7FBDB129C7D"
                        },
                        "E8:61:61:4E:6B:34":    {
                                "rssi": -79,
                                "timestamp":    "1630260478",
                                "data": "0201061BFF9904050FAE590FC85E03A0FE140020ABD603763EE861614E6B34"
                        },
                        "FD:3B:AB:A9:88:BA":    {
                                "rssi": -82,
                                "timestamp":    "1630260474",
                                "data": "0201061BFF9904050F53576CC82C02800328004CAB76C87627FD3BABA988BA"
                        }
                }
        }
}
I (712910) ruuvi_gateway: [main] free heap: 131400
I (722910) ruuvi_gateway: [main] free heap: 130588
E (727795) esp-tls: Failed to connnect to host (errno 113)
E (727796) esp-tls: Failed to open new connection
E (727796) TRANS_SSL: Failed to open a new connection
E (727802) HTTP_CLIENT: Connection failed, sock < 0
E (727807) http: [adv_post_task] ../main/http.c:165 {http_send}: HTTP POST to URL=https://network.ruuvi.com/record: request failed, err=28674 (ERROR)
I (727822) LEDS: [adv_post_task] leds_indication_network_no_connection
I (727828) LEDS: [adv_post_task] LED: Start blinking, interval: 200 ms, duty cycle: 50%
I (732910) ruuvi_gateway: [main] free heap: 137240
I (742910) ruuvi_gateway: [main] free heap: 137112
I (752910) ruuvi_gateway: [main] free heap: 137364
I (762090) wifi:state: run -> init (2c0)
I (762091) wifi:pm stop, total sleep time: 0 us / 755328395 us

I (762092) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (762096) wifi_manager: [sys_evt] WIFI_EVENT_STA_DISCONNECTED
I (762104) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 2
I (762111) wifi_manager: [wifi_manager] lost connection
I (762117) wifi_manager: [wifi_manager] Set STA IP String to: 0.0.0.0
W (762124) ruuvi_gateway: [wifi_manager] Wifi disconnected
I (762130) LEDS: [wifi_manager] leds_indication_network_no_connection
I (762137) LEDS: [wifi_manager] LED: Start blinking, interval: 200 ms, duty cycle: 50%
I (762147) wifi_manager: [wifi_manager] MESSAGE: ORDER_CONNECT_STA
I (762154) TIME: [time_task] Deactivate SNTP time synchronization
I (762245) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (762246) wifi:state: init -> auth (b0)
I (762910) ruuvi_gateway: [main] free heap: 138052
I (763247) wifi:state: auth -> init (200)
I (763248) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (763249) wifi_manager: [sys_evt] WIFI_EVENT_STA_DISCONNECTED
I (763253) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 2
I (763262) wifi_manager: [wifi_manager] lost connection
I (763268) wifi_manager: [wifi_manager] Set STA IP String to: 0.0.0.0
I (763275) wifi_manager: [wifi_manager] MESSAGE: ORDER_CONNECT_STA
I (763371) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (763373) wifi:state: init -> auth (b0)
I (764373) wifi:state: auth -> init (200)
I (764374) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (764375) wifi_manager: [sys_evt] WIFI_EVENT_STA_DISCONNECTED
I (764380) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 2
I (764388) wifi_manager: [wifi_manager] lost connection
I (764394) wifi_manager: [wifi_manager] Set STA IP String to: 0.0.0.0
I (764401) wifi_manager: [wifi_manager] MESSAGE: ORDER_CONNECT_STA
I (764498) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (764499) wifi:state: init -> auth (b0)
I (764829) wifi:state: auth -> assoc (0)
I (765830) wifi:state: assoc -> init (400)
I (765831) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (765831) wifi_manager: [sys_evt] WIFI_EVENT_STA_DISCONNECTED
I (765837) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 4
I (765845) wifi_manager: [wifi_manager] lost connection
I (765851) wifi_manager: [wifi_manager] Set STA IP String to: 0.0.0.0
I (765858) wifi_manager: [wifi_manager] MESSAGE: ORDER_CONNECT_STA
I (765931) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (765932) wifi:state: init -> auth (b0)

```

[mokkilogi-release181-wifi-beta-update-interval copy.txt](https://github.com/ruuvi/ruuvi.gateway_esp.c/files/7073165/mokkilogi-release181-wifi-beta-update-interval.copy.txt)

**Comment by laurijamsa** [comment]
Interesting. Last night the connection recovered some times and then died again. Some log snippets below and a full log here: 

[mokkilogi-release181-wifi-beta-update-interval copy 2.txt](https://github.com/ruuvi/ruuvi.gateway_esp.c/files/7074150/mokkilogi-release181-wifi-beta-update-interval.copy.2.txt)


```I (7561658) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (7561659) wifi_manager: [sys_evt] WIFI_EVENT_STA_DISCONNECTED
I (7561664) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 2
I (7561672) wifi_manager: [wifi_manager] lost connection
I (7561679) wifi_manager: [wifi_manager] Set STA IP String to: 0.0.0.0
I (7561686) wifi_manager: [wifi_manager] MESSAGE: ORDER_CONNECT_STA
I (7561782) wifi:new:<9,0>, old:<9,0>, ap:<255,255>, sta:<9,0>, prof:1
I (7561783) wifi:state: init -> auth (b0)
I (7562165) wifi:state: auth -> assoc (0)
I (7562358) wifi:state: assoc -> run (10)
I (7562910) ruuvi_gateway: [main] free heap: 137792
I (7563447) wifi:connected with Secret Network 2G, aid = 2, channel 9, BW20, bssid = 00:1e:42:32:c1:62
I (7563447) wifi:security: WPA2-PSK, phy: bgn, rssi: -71
I (7563451) wifi:pm start, type: 0
I (7563454) wifi_manager: [sys_evt] WIFI_EVENT_STA_CONNECTED
I (7563521) wifi:AP's beacon interval = 102400 us, DTIM period = 2
I (7762910) ruuvi_gateway: [main] free heap: 137920
I (7772910) ruuvi_gateway: [main] free heap: 137920
I (7824256) wifi_manager: [sys_evt] IP_EVENT_STA_GOT_IP
I (7824257) tcpip_adapter: sta ip: 192.168.1.229, mask: 255.255.255.0, gw: 192.168.1.1
I (7824260) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_GOT_IP
I (7824267) wifi_manager: [wifi_manager] About to save config to flash
D (7824275) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:302 {wifi_sta_config_save}: wifi_sta_config: ssid:Sect
D (7824291) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:303 {wifi_sta_config_save}: wifi_settings: SoftAP_ssi 
D (7824305) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:304 {wifi_sta_config_save}: wifi_settings: SoftAP_pwd8
D (7824319) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:305 {wifi_sta_config_save}: wifi_settings: SoftAP_cha1
D (7824333) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:306 {wifi_sta_config_save}: wifi_settings: SoftAP_hid0
D (7824348) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:307 {wifi_sta_config_save}: wifi_settings: SoftAP_ban1
D (7824364) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:308 {wifi_sta_config_save}: wifi_settings: sta_only (1
D (7824381) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:309 {wifi_sta_config_save}: wifi_settings: sta_power_0
D (7824396) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:310 {wifi_sta_config_save}: wifi_settings: sta_static0
D (7824412) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:311 {wifi_sta_config_save}: wifi_settings: sta_ip_add0
D (7824427) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:312 {wifi_sta_config_save}: wifi_settings: sta_gw_add0
D (7824441) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_sta_config.c:313 {wifi_sta_config_save}: wifi_settings: sta_netmas0
I (7824456) wifi_manager: [wifi_manager] Set STA IP String to: 192.168.1.229
I (7824463) ruuvi_gateway: [wifi_manager] Wifi connected
I (7824469) LEDS: [wifi_manager] leds_indication_on_network_ok
I (7824476) LEDS: [wifi_manager] LED: OFF
I (7824481) TIME: [time_task] Activate SNTP time synchronization
I (7832910) ruuvi_gateway: [main] free heap: 137672
I (7842910) ruuvi_gateway: [main] free heap: 137672
W (7845667) wifi:ba not setup
W (7845669) wifi:ba not setup
W (7845670) wifi:ba not setup
W (7845848) wifi:token mismatch, expect=4
W (7845850) wifi:token mismatch, expect=4
W (7846031) wifi:ba not setup
W (7846212) wifi:ba not setup
W (7846398) wifi:ba not setup

I (7855726) TIME: [tiT] Time has been synchronized: 2021-08-29 20:07:07.267
I (7862910) ruuvi_gateway: [main] free heap: 91624
W (7863228) esp-tls: Failed to open new connection in specified timeout
E (7863228) TRANS_SSL: Failed to open a new connection
E (7863233) HTTP_CLIENT: Connection failed, sock < 0
E (7863236) http: [adv_post_task] ../main/http.c:165 {http_send}: HTTP POST to URL=https://network.ruuvi.com/record: request failed, err=28674 (ERROR)
I (7863250) LEDS: [adv_post_task] leds_indication_network_no_connection
I (7863257) LEDS: [adv_post_task] LED: Start blinking, interval: 200 ms, duty cycle: 50%
I (7872910) ruuvi_gateway: [main] free heap: 135524

I (37672418) http: [adv_post_task] HTTP POST to URL=https://network.ruuvi.com/record: STATUS=200
I (37672421) LEDS: [adv_post_task] leds_indication_on_network_ok
I (37672423) LEDS: [adv_post_task] LED: OFF
I (37672910) ruuvi_gateway: [main] free heap: 136392
I (37682910) ruuvi_gateway: [main] free heap: 136908
I (37692910) ruuvi_gateway: [main] free heap: 136908
I (37702910) ruuvi_gateway: [main] free heap: 136908
I (37712910) ruuvi_gateway: [main] free heap: 136908
I (37722910) ruuvi_gateway: [main] free heap: 136908
I (37726665) ADV_POST_TASK: [adv_post_task] Advertisements in table: 8
I (37726666) ADV_POST_TASK: [adv_post_task] i: 0, tag: F6:07:52:34:28:C6, rssi: -67, timestamp: 1630297497:
I (37726672) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E A5 51 F7 C8 AF 03 E0 | ..........Q.....
I (37726682) ADV_POST_TASK: [adv_post_task] 0010: FF 10 FF D4 AA F6 21 F0 5C F6 07 52 34 28 C6    | ......!.\..R4(.
I (37726692) ADV_POST_TASK: [adv_post_task] i: 1, tag: CF:C6:30:48:71:CE, rssi: -56, timestamp: 1630297497:
I (37726703) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0F 21 4D BD 0E 3A 00 4C | .........!M..:.L
I (37726713) ADV_POST_TASK: [adv_post_task] 0010: FF B0 03 E0 9E 36 81 91 F6 CF C6 30 48 71 CE    | .....6.....0Hq.
I (37726723) ADV_POST_TASK: [adv_post_task] i: 2, tag: F0:4C:E2:AA:3F:13, rssi: -78, timestamp: 1630297497:
I (37726733) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E 1A 53 90 C9 18 00 04 | ..........S.....
I (37726744) ADV_POST_TASK: [adv_post_task] 0010: FF C4 04 14 AF 36 CF BE B2 F0 4C E2 AA 3F 13    | .....6....L..?.
I (37726754) ADV_POST_TASK: [adv_post_task] i: 3, tag: E5:06:6D:26:1F:9C, rssi: -74, timestamp: 1630297497:
I (37726764) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0F 3A 4E A0 16 61 00 14 | .........:N..a..
I (37726774) ADV_POST_TASK: [adv_post_task] 0010: FF D8 04 14 AD 96 7D 92 2C E5 06 6D 26 1F 9C    | ......}.,..m&..
I (37726784) ADV_POST_TASK: [adv_post_task] i: 4, tag: F8:78:29:D9:1B:80, rssi: -86, timestamp: 1630297497:
I (37726795) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E 0D 59 FA C8 E7 04 04 | ..........Y.....
I (37726805) ADV_POST_TASK: [adv_post_task] 0010: FF 0C 00 18 AA 16 14 AE 95 F8 78 29 D9 1B 80    | ..........x)...
I (37726815) ADV_POST_TASK: [adv_post_task] i: 5, tag: E8:61:61:4E:6B:34, rssi: -78, timestamp: 1630297496:
I (37726826) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0F 00 58 1B C9 0A 03 A4 | ..........X.....
I (37726836) ADV_POST_TASK: [adv_post_task] 0010: FE 1C 00 20 AA 96 03 AE 83 E8 61 61 4E 6B 34    | ... ......aaNk4
I (37726846) ADV_POST_TASK: [adv_post_task] i: 6, tag: D7:FB:DB:12:9C:7D, rssi: -89, timestamp: 1630297496:
I (37726857) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 10 29 56 26 C8 B0 03 F8 | .........)V&....
I (37726867) ADV_POST_TASK: [adv_post_task] 0010: FE F4 FF B4 AB D6 5D A7 4F D7 FB DB 12 9C 7D    | ......].O.....}
I (37726877) ADV_POST_TASK: [adv_post_task] i: 7, tag: DF:AA:90:34:69:05, rssi: -85, timestamp: 1630297491:
I (37726887) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E 12 52 E9 C8 55 00 1C | ..........R..U..
I (37726898) ADV_POST_TASK: [adv_post_task] 0010: FF CC 04 08 B2 D6 C4 DD 0F DF AA 90 34 69 05    | ............4i.
I (37726915) http: [adv_post_task] HTTP POST to URL=https://network.ruuvi.com/record, DATA:
{
        "data": {
                "coordinates":  "",
                "timestamp":    "1630297498",
                "nonce":        "1862048500",
                "gw_mac":       "EE:39:AD:AB:D4:E2",
                "tags": {
                        "F6:07:52:34:28:C6":    {
                                "rssi": -67,
                                "timestamp":    "1630297497",
                                "data": "0201061BFF9904050EA551F7C8AF03E0FF10FFD4AAF621F05CF607523428C6"
                        },
                        "CF:C6:30:48:71:CE":    {
                                "rssi": -56,
                                "timestamp":    "1630297497",
                                "data": "0201061BFF9904050F214DBD0E3A004CFFB003E09E368191F6CFC6304871CE"
                        },
                        "F0:4C:E2:AA:3F:13":    {
                                "rssi": -78,
                                "timestamp":    "1630297497",
                                "data": "0201061BFF9904050E1A5390C9180004FFC40414AF36CFBEB2F04CE2AA3F13"
                        },
                        "E5:06:6D:26:1F:9C":    {
                                "rssi": -74,
                                "timestamp":    "1630297497",
                                "data": "0201061BFF9904050F3A4EA016610014FFD80414AD967D922CE5066D261F9C"
                        },
                        "F8:78:29:D9:1B:80":    {
                                "rssi": -86,
                                "timestamp":    "1630297497",
                                "data": "0201061BFF9904050E0D59FAC8E70404FF0C0018AA1614AE95F87829D91B80"
                        },
                        "E8:61:61:4E:6B:34":    {
                                "rssi": -78,
                                "timestamp":    "1630297496",
                                "data": "0201061BFF9904050F00581BC90A03A4FE1C0020AA9603AE83E861614E6B34"
                        },
                        "D7:FB:DB:12:9C:7D":    {
                                "rssi": -89,
                                "timestamp":    "1630297496",
                                "data": "0201061BFF99040510295626C8B003F8FEF4FFB4ABD65DA74FD7FBDB129C7D"
                        },
                        "DF:AA:90:34:69:05":    {
                                "rssi": -85,
                                "timestamp":    "1630297491",
                                "data": "0201061BFF9904050E1252E9C855001CFFCC0408B2D6C4DD0FDFAA90346905"
                        }
                }
        }
}
I (37732910) ruuvi_gateway: [main] free heap: 92500
W (37734768) esp-tls: Failed to open new connection in specified timeout
E (37734768) TRANS_SSL: Failed to open a new connection
E (37734774) HTTP_CLIENT: Connection failed, sock < 0
E (37734776) http: [adv_post_task] ../main/http.c:165 {http_send}: HTTP POST to URL=https://network.ruuvi.com/record: request failed, err=28674 (ERRO)
I (37734791) LEDS: [adv_post_task] leds_indication_network_no_connection
I (37734798) LEDS: [adv_post_task] LED: Start blinking, interval: 200 ms, duty cycle: 50%
I (37742910) ruuvi_gateway: [main] free heap: 137124
I (37752910) ruuvi_gateway: [main] free heap: 137124
I (37762910) ruuvi_gateway: [main] free heap: 137124
I (37772910) ruuvi_gateway: [main] free heap: 137124
I (37782910) ruuvi_gateway: [main] free heap: 137124
I (37786665) ADV_POST_TASK: [adv_post_task] Advertisements in table: 8
I (37786666) ADV_POST_TASK: [adv_post_task] i: 0, tag: E8:61:61:4E:6B:34, rssi: -75, timestamp: 1630297557:
I (37786672) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E FE 58 0C C9 08 03 A4 | ..........X.....
I (37786682) ADV_POST_TASK: [adv_post_task] 0010: FE 20 00 10 AA 96 03 AE 9A E8 61 61 4E 6B 34    | . ........aaNk4
I (37786692) ADV_POST_TASK: [adv_post_task] i: 1, tag: F6:07:52:34:28:C6, rssi: -66, timestamp: 1630297557:
I (37786703) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E A6 52 02 C8 B0 03 E8 | ..........R.....
I (37786713) ADV_POST_TASK: [adv_post_task] 0010: FF 10 FF D4 AC 56 21 F0 73 F6 07 52 34 28 C6    | .....V!.s..R4(.
I (37786723) ADV_POST_TASK: [adv_post_task] i: 2, tag: CF:C6:30:48:71:CE, rssi: -53, timestamp: 1630297557:
I (37786733) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0F 21 4D B6 0E 3C 00 48 | .........!M..<.H
I (37786744) ADV_POST_TASK: [adv_post_task] 0010: FF A8 03 E4 9E B6 81 92 0E CF C6 30 48 71 CE    | ...........0Hq.
I (37786754) ADV_POST_TASK: [adv_post_task] i: 3, tag: E5:06:6D:26:1F:9C, rssi: -70, timestamp: 1630297557:
I (37786764) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0F 41 4E 8B 16 62 00 10 | .........AN..b..
I (37786774) ADV_POST_TASK: [adv_post_task] 0010: FF E0 04 18 AD 36 7D 92 43 E5 06 6D 26 1F 9C    | .....6}.C..m&..
I (37786784) ADV_POST_TASK: [adv_post_task] i: 4, tag: F0:4C:E2:AA:3F:13, rssi: -76, timestamp: 1630297557:
I (37786795) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E 1E 53 91 C9 16 00 08 | ..........S.....
I (37786805) ADV_POST_TASK: [adv_post_task] 0010: FF C8 04 18 B0 B6 CF BE E1 F0 4C E2 AA 3F 13    | ..........L..?.
I (37786815) ADV_POST_TASK: [adv_post_task] i: 5, tag: D7:FB:DB:12:9C:7D, rssi: -84, timestamp: 1630297556:
I (37786826) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 10 2B 56 0B C8 AD 03 F8 | .........+V.....
I (37786836) ADV_POST_TASK: [adv_post_task] 0010: FE F8 FF BC AB D6 5D A7 67 D7 FB DB 12 9C 7D    | ......].g.....}
I (37786846) ADV_POST_TASK: [adv_post_task] i: 6, tag: DF:AA:90:34:69:05, rssi: -87, timestamp: 1630297542:
I (37786857) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E 13 52 EE C8 54 00 18 | ..........R..T..
I (37786867) ADV_POST_TASK: [adv_post_task] 0010: FF CC 04 04 B4 36 C4 DD 23 DF AA 90 34 69 05    | .....6..#...4i.
I (37786877) ADV_POST_TASK: [adv_post_task] i: 7, tag: F8:78:29:D9:1B:80, rssi: -76, timestamp: 1630297556:
I (37786887) ADV_POST_TASK: [adv_post_task] 0000: 02 01 06 1B FF 99 04 05 0E 0C 5A 10 C8 E8 04 04 | ..........Z.....
I (37786898) ADV_POST_TASK: [adv_post_task] 0010: FF 08 00 24 A9 B6 14 AE AC F8 78 29 D9 1B 80    | ...$......x)...
I (37786916) http: [adv_post_task] HTTP POST to URL=https://network.ruuvi.com/record, DATA:
{
        "data": {
                "coordinates":  "",
                "timestamp":    "1630297558",
                "nonce":        "1862048501",
                "gw_mac":       "EE:39:AD:AB:D4:E2",
                "tags": {
                        "E8:61:61:4E:6B:34":    {
                                "rssi": -75,
                                "timestamp":    "1630297557",
                                "data": "0201061BFF9904050EFE580CC90803A4FE200010AA9603AE9AE861614E6B34"
                        },
                        "F6:07:52:34:28:C6":    {
                                "rssi": -66,
                                "timestamp":    "1630297557",
                                "data": "0201061BFF9904050EA65202C8B003E8FF10FFD4AC5621F073F607523428C6"
                        },
                        "CF:C6:30:48:71:CE":    {
                                "rssi": -53,
                                "timestamp":    "1630297557",
                                "data": "0201061BFF9904050F214DB60E3C0048FFA803E49EB681920ECFC6304871CE"
                        },
                        "E5:06:6D:26:1F:9C":    {
                                "rssi": -70,
                                "timestamp":    "1630297557",
                                "data": "0201061BFF9904050F414E8B16620010FFE00418AD367D9243E5066D261F9C"
                        },
                        "F0:4C:E2:AA:3F:13":    {
                                "rssi": -76,
                                "timestamp":    "1630297557",
                                "data": "0201061BFF9904050E1E5391C9160008FFC80418B0B6CFBEE1F04CE2AA3F13"
                        },
                        "D7:FB:DB:12:9C:7D":    {
                                "rssi": -84,
                                "timestamp":    "1630297556",
                                "data": "0201061BFF990405102B560BC8AD03F8FEF8FFBCABD65DA767D7FBDB129C7D"
                        },
                        "DF:AA:90:34:69:05":    {
                                "rssi": -87,
                                "timestamp":    "1630297542",
                                "data": "0201061BFF9904050E1352EEC8540018FFCC0404B436C4DD23DFAA90346905"
                        },
                        "F8:78:29:D9:1B:80":    {
                                "rssi": -76,
                                "timestamp":    "1630297556",
                                "data": "0201061BFF9904050E0C5A10C8E80404FF080024A9B614AEACF87829D91B80"
                        }
                }
        }
}
I (37792463) http: [adv_post_task] HTTP POST to URL=https://network.ruuvi.com/record: STATUS=200
I (37792466) LEDS: [adv_post_task] leds_indication_on_network_ok
I (37792468) LEDS: [adv_post_task] LED: OFF
I (37792910) ruuvi_gateway: [main] free heap: 136804
I (37802910) ruuvi_gateway: [main] free heap: 137124
I (37812910) ruuvi_gateway: [main] free heap: 137124
I (37822910) ruuvi_gateway: [main] free heap: 137124

```

**Comment by laurijamsa** [comment]
Here is a new log after a full power off reset. It contains interesting lines like:

```
W (25131) wifi:token mismatch, expect=9

W (94962) wifi:ba not setup
```

[mokkilogi-release181-wifi-beta-update-interval-reset copy.txt](https://github.com/ruuvi/ruuvi.gateway_esp.c/files/7075128/mokkilogi-release181-wifi-beta-update-interval-reset.copy.txt)

**Comment by TheSomeMan** [comment]
maybe this is because of limiting Wi-Fi max TX power?: https://github.com/ruuvi/ruuvi.gateway_esp.c/issues/308

**Comment by ojousima** [comment]
Cannot reproduec, waiting for user feedback

**Comment by TheSomeMan** [comment]
As far as I can see, there are two reasons for the disconnections - 3 and 201:
```
I (38584182) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 3
```
```
I (38586283) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 201
```

```
    WIFI_REASON_AUTH_LEAVE               = 3,
    WIFI_REASON_NO_AP_FOUND           = 201,
```

**Comment by laurijamsa** [comment]
Also reason code 4 is pretty famous on the log file.

**Comment by laurijamsa** [comment]
And reason code 2.

**Comment by TheSomeMan** [comment]
```
    WIFI_REASON_AUTH_EXPIRE              = 2,
    WIFI_REASON_AUTH_LEAVE               = 3,
    WIFI_REASON_ASSOC_EXPIRE             = 4,
    WIFI_REASON_NO_AP_FOUND              = 201,
```

**Comment by io53** [comment]
Guess I talked too soon, I notice one gateway disconnect from wifi today. I might have missed it before as gateways shows up as "LIFX lightning" in unifi alerts and I've had one connected to ethernet so I always got data, but I'm quite sure this have not happened in earlier versions. Sadly my network statistics retention is set too short so I cannot see more info from there other than that it has not happened more than once the last 24h. I'll start logging the output and see if I can catch some codes..

**Comment by ojousima** [comment]
This might be fixed with SDK update related to 1.10.0, #448 . Let's give it a throughout testing.

**Comment by ojousima** [comment]
@laurijamsa @io53 Can you check if this is still a problem in v1.10.1-RC1?

**Comment by laurijamsa** [comment]
I don't have access to my Gateway right now but I will try to check this later. Self-assigned.

**Comment by laurijamsa** [comment]
No updates from my side. Haven't had a chance to test the Wi-Fi.

**Comment by ojousima** [comment]
Should be fixed in #518, let's reopen if issue persists
